File: src/Net/Store/Before/Employee.cs
namespace Before
{
    using System.Collections.Generic;
    using System.Linq;

    public abstract class Employee
    {
        public string FirstName { get; private set; }

        public string LastName { get; private set; }

        public decimal FixedSalary { get; private set; }

        public Employee Manager { get; internal set; }

        public string Street { get; set; }

        public string City { get; set; }

        public string Country { get; set; }

        /*
         * Only the managers have subordinates
         * but anyone could have a manager.
         */
        protected IList<Employee> subordinates = new List<Employee>();

        public IEnumerable<Employee> Subordinates
        {
            get { return subordinates.ToArray(); }
        }

        protected Employee(string firstName, string lastName, decimal fixedSalary)
        {
            this.FirstName = firstName;
            this.LastName = lastName;
            this.FixedSalary = fixedSalary;
        }

        public void AddSubordinate(Employee subordinate)
        {
            subordinates.Add(subordinate);
            subordinate.Manager = this;
        }

        public void RemoveSubordinate(Employee subordinate)
        {
            subordinates.Remove(subordinate);
            subordinate.Manager = null;
        }
    }
}

File: src/Java/1. First Course/0. Initial Tee/src/main/refactoringgolf/store/Employee.java
package refactoringgolf.store;

import java.util.Collections;
import java.util.HashSet;
import java.util.Set;

public abstract class Employee {

	protected String firstName;
	protected String lastName;
	protected float fixedSalary;
	protected Employee manager;
	protected String street;
	protected String city;
	protected String country;
	protected Set<Employee> subordinates = new HashSet<Employee>();

	protected Employee(String firstName, String lastName, float fixedSalary) {
		this.firstName = firstName;
		this.lastName = lastName;
		this.fixedSalary = fixedSalary;
	}

	public String getFirstName() {
		return firstName;
	}

	public String getLastName() {
		return lastName;
	}

	public float getFixedSalary() {
		return fixedSalary;
	}

	public Employee getManager() {
		return manager;
	}

	public String getStreet() {
		return street;
	}

	public void setStreet(String street) {
		this.street = street;
	}

	public String getCity() {
		return city;
	}

	public void setCity(String city) {
		this.city = city;
	}

	public String getCountry() {
		return country;
	}

	public void setCountry(String country) {
		this.country = country;
	}

	public Set<Employee> getSubordinates() {
		return Collections.unmodifiableSet(subordinates);
	}

	public void addSubordinate(Employee subordinate) {
		subordinates.add(subordinate);
		subordinate.manager = this;
	}

	public void removeSubordinate(Employee subordinate) {
		subordinates.remove(subordinate);
		subordinate.manager = null;
	}
}
